com.microsoft.java.debug.plugin/src/main/java/com/microsoft/java/debug/plugin/internal/JdtSourceLookUpProvider.java

```diff
@@ -533,4 +571,58 @@ private boolean isSameURI(String uri1, String uri2) {
             return false;
         }
     }
+
+    public int[] getOriginalLineMappings(String uri) {
+        IClassFile classFile = resolveClassFile(uri);
+        try {
+            if (classFile == null) {
+                return null;
+            }
+
+            IPackageFragmentRoot packageRoot = (IPackageFragmentRoot) classFile.getAncestor(IJavaElement.PACKAGE_FRAGMENT_ROOT);
+            if (packageRoot != null && packageRoot.getSourceAttachmentPath() != null) {
+                return null;
+            }
+
+            if (classFile.getSourceRange() == null) {
+                ContentProviderManager contentProvider = JavaLanguageServerPlugin.getContentProviderManager();
+                try {
+                    DecompilerResult result = contentProvider.getSourceResult(classFile, new NullProgressMonitor());
+                    if (result != null) {
+                        return result.getOriginalLineMappings();
+                    }
+                } catch (NoSuchMethodError e) {
```

Comment:
Will the `NoClassDefFoundError` throw here? I remember `DecompilerResult` is newly added as well.

Reply:
> DecompilerResult result = contentProvider.getSourceResult(classFile, new NullProgressMonitor());

`getSourceResult` is also a new method, and will be thrown first.

Reply:
ok